code/game/turfs/turf.dm

```diff
@@ -167,12 +167,13 @@ GLOBAL_LIST_EMPTY(station_turfs)
 	SEND_SIGNAL(src, COMSIG_TURF_MULTIZ_NEW, T, dir)
 
 ///returns if the turf has something dense inside it. if exclude_mobs is true, skips dense mobs like fat yoshi. if exclude_object is true, it will exclude the excluded_object you sent through
-/turf/proc/is_blocked_turf(exclude_mobs, excluded_object)
+/turf/proc/is_blocked_turf(exclude_mobs, list/excluded_objects = list())
 	if(density)
 		return TRUE
+	var/list/excluded_typecache = typecacheof(excluded_objects)
```

Comment:
Loop through the list using `istype()` checks instead of building a typecache every time the proc runs.

Reply:
Removed the typecache